### PR TITLE
footer: Use Native App Install Prompt to install Android app

### PR DESCRIPTION
**Pull request by pwithnall**
When supported, use this API to provide a slightly more streamlined experience for installing the Android app when pressing the “Get it on Google Play” button.

The API is only supported by Chrome, on mobile, if the app is advertised as a related application in the PWA manifest for the site, and if the app is not already installed.

The full list of requirements are here:
https://developer.chrome.com/blog/app-install-banners-native/#what-are-the-criteria

If the API is not supported, the button will function as a link as normal.

Signed-off-by: Philip Withnall <pwithnall@endlessos.org>

https://github.com/endlessm/endless-key-content-private/issues/64

### CODE REVIEW

**Comment by pwithnall** [comment]
I haven’t been able to test this as the API is not supported on the desktop, and on my phone I can’t get Chrome to load a locally-hosted version of the site (and can’t figure out why). Hopefully we can test this using the dev instance.

**Comment by manuq** [comment]
I have added a fixup to satisfy the linter. Done by running `pre-commit run --all-files` locally which runs the code formatters.

File: kolibri_explore_plugin/assets/src/components/AboutFooter.vue [hunk 1]
```diff
@@ -95,6 +101,45 @@
         return plugin_data.androidApplicationId != '' && plugin_data.windowsApplicationId != '';
       },
     },
+    created() {
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/components/AboutFooter.vue:104]: For better component lifecycle let's move the `addEventListener` to the `mounted` stage, and the `removeEventListener` to the `beforeDestroy` stage. See https://v2.vuejs.org/v2/guide/instance.html#Lifecycle-Diagram

File: kolibri_explore_plugin/assets/src/components/AboutFooter.vue [hunk 2]
```diff
@@ -77,6 +78,11 @@
 
   export default {
     name: 'AboutFooter',
+    data() {
+      return {
+        deferredInstallPrompt: null,
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/components/AboutFooter.vue:83]: :+1:

**Review by manuq — CHANGES_REQUESTED** [review]

**Comment by pwithnall** [comment]
Thanks for the helpful review comments. Pushed a couple of fixups for those issues.

**Comment by manuq** [comment]
Thank you @pwithnall, I used the "squash and merge" from github to merge, I hope that's fine.

**Pull request merged**